File: swift/bob/Bob/Bob/Bob.swift
import Foundation

class Bob {
	static func hey(text:String) -> NSString {
		if(text.uppercaseString == text && text.rangeOfString("[A-Z]", options: .RegularExpressionSearch) != nil) {
			return "Woah, chill out!";
		} else if(count(text.stringByTrimmingCharactersInSet(NSCharacterSet.whitespaceAndNewlineCharacterSet())) == 0) {
			return "Fine, be that way.";
		} else if(text.hasSuffix("?")) {
			return "Sure.";
		} else {
			return "Whatever.";
		}
	}
}

File: python/bob/bob.py
import re;

def hey(what):
    what = what.strip()
    if what == '':
        return 'Fine. Be that way!'
    if what == what.upper() and re.match('.*[a-zA-Z]', what) is not None:
        return 'Whoa, chill out!'
    if what[-1] == '?':
        return 'Sure.'
    return 'Whatever.'
